src/database.py

```diff
@@ -21,17 +23,49 @@ class DatabaseConnectionInfo(WithLogging):
     username: str
     password: str
 
-    def verify(self) -> bool:
-        """Verify the PostgreSQL connection."""
+    def execute(self, dbname: str, query: str, vars=None) -> tuple[bool, list]:
```

Comment:
**suggestion** maybe there is a use-case for having the database name as part of the attributes of the class (together with username, password, etc). This was the code would read better I believe, since we centralize these information into a single class. Right now we pass the database name around separately, with risk of inconsistency

Reply:
Actually maybe we can have a dataclass that encapsulate all information: database name, table name, endpoint, etc, that can be passed around. This makes the objects a bit transparent, with a single dependency, instead of overusing constants (that are subtle, not transparent depenencies - less functional)

Reply:
The reason I have not included the database name as an attribute of the dataclass is because the same class might need to connect to multiple databases and not a single database. For example, when creating the authentication database fixture, the charm needs to connect to `auth_db` database, however, when removing the fixture, it should connect to some other database (in this case `postgres`) because one cannot remove the database it is connected to.

In case of `auth`, it makes sense to add `auth_db` as `dbname` property in the dataclass, but we would still need to use `POSTGRESQL_DEFAULT_DATABASE` from the constants.

Reply:
I have updated the `DatabaseConnectionInfo` dataclass to include `dbname` as well. (This will be used by default if `dbname` param is not passed in the `execute` function. I believe that a table name is something that is passed in a query and thus would not have it as an attribute to DatabaseConnectionInfo (because this class should be able to query to / from multiple tables and not only one table.)

Reply:
What about then to have this signature:

```
Authentication(database:DatabaseConnectionInfo, table_name: str)
```

This makes the class "pure", without using external, non-declared and implicit dependency. It should even make it easier to test

Reply:
I'm more comfortable with having these constants as static class properties rather than receiving them as constructor param. This way, the class would not have any external variable reference and this would not require us change other parts in the codebase as well since we keep the same constructor signature. This will kill two birds with a single stone. WDYT @deusebio ?

Reply:
I don't have strong opinion, but If I have to say my preference, I would make the table name configurable and admin to be a class static property. Such that one could even use this manager in tests easily with different tables or abstracted from the charm usage